src/components/ReportActionItemIOUAction.js

```diff
@@ -16,7 +16,7 @@ const propTypes = {
     chatReportID: PropTypes.number.isRequired,
 
     /** Should render the preview Component? */
-    shouldDisplayPreview: PropTypes.bool.isRequired,
+    isMostRecentIOUReportAction: PropTypes.bool.isRequired,
```

Comment:
The reason for this change is that the Component itself is [now deciding whether to display itself or not](https://github.com/Expensify/Expensify.cash/pull/3123/files#diff-3597ac886dade8ed106d403fad7c534b6a0fd9423bf6081877749b0295fb3bc8R88-R91). It seemed confusing to me to have both that logic and this prop. 

As for why the prop shouldn't be handled in the same logic, perhaps it should, but it felt like I was passing down too much state about parents via the props.

Any thoughts on th best solution here?

Reply:
We don't really have strict guidelines here I guess... but I think in general components don't need to decide whether to render if their parents can decide whether they should render. e.g.

```js
{something && <Component />}
...

render() {
    return <Stuff />
}
```

compared to

```js
<Component something={something} />
...

render() {
    if (!something) {
        return null; 
    }

    return <Stuff />
}
```

Reply:
Why is it be better for parents to decide to run the component instead of the component itself? The argument for the component to decide itself is that we don't need to duplicate the logic in each place it is used.

Reply:
>The argument for the component to decide itself is that we don't need to duplicate the logic in each place it is used

Alright I'll pull out another example.... consider a case like this...

https://github.com/Expensify/Expensify.cash/blob/9c265ac32fa8456154f2cc1a534e47605e8ee955/src/components/FullscreenLoadingIndicator.js#L17-L21

Everywhere this component should show now must do this...

```js
// When we want it to show no matter what
<FullScreenLoadingIndicator visible />

// When we only want it to show sometimes
<FullScreenLoadingIndicator visible={this.state.shouldWeShowThisThing} />
```

compare again to

```js
// When we want it to show no matter what
<FullScreenLoadingIndicator />

// When we only want it to show sometimes
{this.state.shouldWeShowThisThing && <FullScreenLoadingIndicator />}
```

There are maybe some exception to this e.g. if your component must perform some task on mount regardless of whether it should render. But I would probably argue against it in most cases where a component is very simple.

Still marked as a NAB since there is no formal guidance on this and it's just my opinion.

Reply:
Oh I see. I agree that if the component itself has no logic (just checks the prop and nothing else), then having it in the parent is better.

Reply:
Cool, thanks for the examples. I have grouped all logic together, parent component now determines whether the Comp should display or not.